extensions/mssql/src/credentialstore/nativeCredentialService.ts

```diff
@@ -0,0 +1,145 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { AppContext } from '../appContext';
+import { SqlOpsDataClient, ISqlOpsFeature, SqlOpsFeature } from 'dataprotocol-client';
+import * as Utils from '../utils';
+import { ClientCapabilities, RPCMessageType, ServerCapabilities } from 'vscode-languageclient';
+import * as Contracts from './contracts';
+import { Disposable, SecretStorage } from 'vscode';
+import * as azdata from 'azdata';
+import * as UUID from 'vscode-languageclient/lib/utils/uuid';
+
+export class NativeCredentialService extends SqlOpsFeature<any> {
```

Comment:
Given that this can handle both the old case (where Linux was storing passwords non-natively) and the new case (keytar) it seems like just `CredentialService` would make more sense here.

Reply:
VScode uses a global service with the name so I wanted to have a separate name for this service

Reply:
I'm not sure what you mean by this - what does what VS Code name their stuff have anything to do with us? At the very least I'd name it `SqlCredentialService` or something - `NativeCredentialService` isn't accurate unless they have the setting enabled so that just seems confusing